sunpy/net/fido_factory.py

```diff
@@ -99,6 +99,7 @@ def _handle_record_slice(self, client_resp, record_slice):
         ret = resp_type(resp)
         # Make sure we pass the client back out again.
         ret.client = client_resp.client
+        ret.query_args = client_resp.query_args
```

Comment:
As the `query_args` attribute is specific to `JSOCResponse` this is the wrong place for this. It should be added in this https://github.com/sunpy/sunpy/blob/master/sunpy/net/jsoc/jsoc.py#L66 `__getitem__`

Reply:
Hi thanks for the review. 

I can add on __getitem__ to get the table and the query_args too. 

But now I have the problem when 

```
# Reconstruct a response object with the sub-indexed records.
ret = resp_type(resp)
```

This do, for some reason, the ret variable have `query_args = None` after `resp_type(resp)`. I don't know how to make the "cast" without erase query_args. 

Any idea?